### PR TITLE
fix: wamp better events sending

### PR DESCRIPTION
**Pull request by xhanin**
includes:
- new reconnect event
- tests for connect / close / reconnect events
- trigger close event on socket reset

### CODE REVIEW

File: ocpp-wamp/src/main/kotlin/com/izivia/ocpp/wamp/client/OcppWampClient.kt [hunk 1]
```diff
@@ -20,13 +20,15 @@ interface OcppWampClient {
             ocppId: CSOcppId,
             ocppVersion: OcppVersion,
             timeoutInMs: Long = 30_000,
+            autoReconnect: Boolean = true,
```
Comment (xhanin) [inline ocpp-wamp/src/main/kotlin/com/izivia/ocpp/wamp/client/OcppWampClient.kt:23]: I needed that for the reconnect tests

File: ocpp-wamp/src/main/kotlin/com/izivia/ocpp/wamp/server/OcppWampServer.kt [hunk 2]
```diff
@@ -62,8 +62,9 @@ interface OcppWampServer {
             path: String = "ws",
             timeoutInMs: Long = 60_000,
             onWsConnectHandler: (CSOcppId, WampMessageMetaHeaders) -> Unit = { _, _ -> },
+            onWsReconnectHandler: (CSOcppId, WampMessageMetaHeaders) -> Unit = { _, _ -> },
```
Comment (xhanin) [inline ocpp-wamp/src/main/kotlin/com/izivia/ocpp/wamp/server/OcppWampServer.kt:65]: this breaks api compatiblity with non named arguments, but considering limited use of the api, I think it's ok. Should be published as a 1.1.0 to warn about that

File: ocpp-wamp/src/main/kotlin/com/izivia/ocpp/wamp/server/impl/OcppWampServerApp.kt [hunk 3]
```diff
@@ -253,3 +274,6 @@ class OcppWampServerApp(
         }
     }
 }
+
+private fun Throwable.isConnectionReset(): Boolean =
+    this is SocketException && this.message.equals("Connection reset")
```
Comment (xhanin) [inline ocpp-wamp/src/main/kotlin/com/izivia/ocpp/wamp/server/impl/OcppWampServerApp.kt:279]: I've tested that with real life case with toxyproxy.
no unit tests with mock it doesn't make much sense

File: ocpp-wamp/src/main/kotlin/com/izivia/ocpp/wamp/server/impl/UndertowWebSocketCallBack.kt [hunk 4]
```diff
@@ -66,7 +66,7 @@ class UndertowWebSocketCallBack(private val ws: WsHandler) : WebSocketConnection
 
         channel.addCloseTask {
             val status = WsStatus(it.closeCode, it.closeReason ?: "unknown")
-            logger.warn("$logContext - closing websocket: WsStatus $status channel $it", Exception("capture stack"))
+            logger.warn("$logContext - closing websocket: WsStatus $status channel $it")
```
Comment (xhanin) [inline ocpp-wamp/src/main/kotlin/com/izivia/ocpp/wamp/server/impl/UndertowWebSocketCallBack.kt:69]: remove that stack, doesn't help

**Review by walien — APPROVED** [review]

**Pull request merged**